src/inmanta/data/__init__.py

```diff
@@ -484,6 +484,49 @@ async def get_list(
         result = await cls.select_query(sql_query, values, no_obj=no_obj, connection=connection)
         return result
 
+    @classmethod
+    async def get_list_paged(
+        cls: Type[TBaseDocument],
+        order_by_column: str,
+        order: str = "ASC",
+        limit: Optional[int] = None,
+        start: Optional[Any] = None,
+        end: Optional[Any] = None,
+        no_obj: bool = False,
+        connection: Optional[asyncpg.connection.Connection] = None,
+        **query: Any,
+    ) -> List[TBaseDocument]:
+        """
+        Get a list of documents matching the filter args, with paging support
+
+        :param order_by_column: The name of the column in the database the sorting should be based on
+        :param order: The order to apply to the sorting
+        :param limit: If specified, the maximum number of entries to return
+        :param start: A value conforming the sorting column type, all returned rows will have greater value in the sorted column
+        :param end: A value conforming the sorting column type, all returned rows will have lower value in the sorted column
+        :param no_obj: Whether not to cast the query result into a matching object
+        :param connection: An optional connection
+        :param **query: Any additonnal filter to apply
+        """
+        query = cls._convert_field_names_to_db_column_names(query)
```

Comment:
Assertions should rather be used for testing right? I can throw an exception though.

Reply:
Actually, why is it that `end`, `start` and `limit` can not be set together? I find it less of a problem than if `start` and `end` are set in a way that the response would exceed `DBLIMIT` entries.  (Which for now is handled by setting the limit to `DBLIMIT` if the request hasn't provided a value, so it is kind of weird to say that it can't be done)

Reply:
True. And we can never enforce that the provided filter happens based on a column with an index. You can ignore this comment.

Reply:
Should I removed the raised `BadRequest` if we specify those three then?

Reply:
> Should I removed the raised `BadRequest` if we specify those three then?

I am not sure. The `get_reports` API all also uses pagination and that one also return an error when `start`, `end` and `limit` start set together. In my opinion, it's important to keep the behavior of all API calls consistent. So I think we should keep it. @wouterdb What do you think.

Reply:
Yes, but that call used to respond with a code 500 for such error, I changed that, so I could also remove it completely if we decide to do it for `get_agents` and `get_agent_processes` right?

Reply:
perhaps we indeed should allow all three at the same time everywhere.

Reply:
Did it in https://github.com/inmanta/inmanta/pull/2521/commits/6ae86c8805d7d303a9499161febe9feb6675aaef